subsys/portability/cmsis_rtos_v1/cmsis_thread.c

```diff
@@ -197,6 +197,7 @@ osStatus osThreadTerminate(osThreadId thread_id)
 	}
 
 	k_thread_abort((k_tid_t)thread_id);
+	z_thread_cmsis_status_mask_clear((k_tid_t)thread_id);
```

Comment:
1. If the thread self-terminates through osThreadTerminate, then the status mask will never be cleared. As a result, that thread will forever be seen as in use.
2. If there is a context switch between returning from `k_thread_abort()` and `z_thread_cmsis_status_mask_clear()`, the aborted thread will not be able to be reused until we context switch back to the original thread again.

I don't know enough about CMSIS to say for certain if either of the above scenarios would be an issue for CMSIS, but I wanted to bring it to attention in case that they might be.

Reply:
hmm, need to check, but add this line does the trick, without it, the cmsis v1 tests fail.

Reply:
I did it differently now